File: dotnet/DesignPatterns/StructuralDesignPatterns/AdapterDesignPattern/AdapterDesignPattern.Example1/Adapters/StripeAdapter.cs
using AdapterDesignPattern.Example1.Interfaces;
using AdapterDesignPattern.Example1.Services;

namespace AdapterDesignPattern.Example1.Adapters;

public class StripeAdapter : IPaymentProcessor
{
	private readonly StripeAPI _stripeAPI;

	public StripeAdapter(StripeAPI stripeAPI)
	{
		_stripeAPI = stripeAPI;
	}

	public void ProcessPayment(double amount, string currency)
	{
		var amountCents = (int)(amount * 100);
		_stripeAPI.Charge(amountCents, currency);
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/AdapterDesignPattern/AdapterDesignPattern.Example1/src/adapters/StripeAdapter.ts
import { IPaymentProcessor } from '../interfaces/IPaymentProcessor';
import { StripeAPI } from '../services/StripeAPI';

export class StripeAdapter implements IPaymentProcessor {
  constructor(private readonly stripeAPI: StripeAPI) {}

  processPayment(amount: number, currency: string): void {
    const amountCents = Math.round(amount * 100);
    this.stripeAPI.charge(amountCents, currency);
  }
}
